src/pip/_internal/cli/base_command.py

```diff
@@ -195,7 +196,8 @@ def _main(self, args):
             logger.debug('Exception information:', exc_info=True)
 
             return PREVIOUS_BUILD_DIR_ERROR
-        except (InstallationError, UninstallationError, BadCommand) as exc:
+        except (InstallationError, UninstallationError,
+                BadCommand, NetworkConnectionError) as exc:
```

Comment:
Probably needs a functional test to make sure this is caught correctly without traceback (unlike the previous `HTTPError`).

Reply:
I am not sure how do I write the functional test. I wrote the following test
```
@mock.patch("pip._internal.network.download.raise_for_status")
def test_install_exits_when_network_error(mock_raise_for_status, script):
    mock_raise_for_status.side_effect = NetworkConnectionError("Http Error")
    result = script.pip('install', 'INITools==0.2', expect_error=True)
    assert result.returncode == ERROR
```

However, it doesn't return any error. Can you guide me here? Although I have also tried to use `caplog` in order to get the logging but that doesn't work either.

Reply:
Ah, ok, this won’t work because you cannot monkey-patch the pip inside `script`. I don’t have a better idea, let’s not have a functional test then.

Reply:
No issue then.